src/Featurizers/Components/StatisticsMetricsEstimator.h

```diff
@@ -0,0 +1,391 @@
+template <typename T>
+class BasicStatsTrainingOnlyPolicy {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using InputType                         = T;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    static constexpr char const * const     NameValue = StatisticalMetricsEstimatorName;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    BasicStatsTrainingOnlyPolicy(void);
+
+    void fit(InputType const &input);
+    BasicStatisticalAnnotationData<T> complete_training(void);
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    InputType                                 _min;
+    InputType                                 _max;
+    std::uint64_t                             _count;
+    bool                                      _first_element_flag;
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         StandardStatsTrainingOnlyPolicy
+///  \brief         StandardStatsTrainingOnlyPolicy deals with advanced stats include
+///                 sum and average
+///
+template <typename T>
+class StandardStatsTrainingOnlyPolicy : public BasicStatsTrainingOnlyPolicy<T> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using InputType                         = T;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    static constexpr char const * const     NameValue = StatisticalMetricsEstimatorName;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    StandardStatsTrainingOnlyPolicy(void);
+
+    void fit(InputType const &input);
+    StandardStatisticalAnnotaionData<T> complete_training(void);
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    long double                               _sum;
+    std::double_t                             _average;

```

Comment:
This value doesn't need to be a member, as it is only used when creating the annotation data.